zio-http/src/main/scala/zio/http/Handler.scala

```diff
@@ -834,6 +862,15 @@ object Handler {
   def tooLarge: Handler[Any, Nothing, Any, Response] =
     Handler.status(Status.RequestEntityTooLarge)
 
+  val unit: Handler[Any, Nothing, Any, Unit] =
+    fromExit(Exit.unit)
+
+  /**
+   * Constructs a handler from a function that uses a web socket.
+   */
+  final def webSocket[Env](f: WebSocketChannel => ZIO[Env, Throwable, Any]): SocketApp[Env] =
```

Comment:
You can be polymorphic on the type of the ZIO effect to preserve it during the handler. In fact because we have monadic composition on handlers, you could use that to achieve a kind of composition of a slightly generalized socket app (with a success type parameter).

```suggestion
  final def webSocket[Env, A](f: WebSocketChannel => ZIO[Env, Throwable, Any]): Handler[Env, Throwable, WebChannelEvent, A] =
```